dadi/lib/controller/domain.js

```diff
@@ -0,0 +1,139 @@
+const path = require('path')
+const DomainManager = require(path.join(__dirname, '/../models/domain-manager'))
+const help = require(path.join(__dirname, '/../help'))
+
+/**
+ * Accept POST requests for adding domains to the internal domain configuration.
+ */
+module.exports.post = (req, res) => {
+  // Don't accept an empty POST
+  if (!req.body || !Array.isArray(req.body) || req.body.length === 0) {
+    return help.sendBackJSON(400, {
+      success: false,
+      errors: ['Bad Request']
+    }, res)
+  }
+
+  let domains = req.body
+
+  domains.forEach((item, index) => {
+    if (!DomainManager.getDomain(item.domain)) {
+      // Prepare the domain configuration.
+      let configContent = {
+        images: {
+          directory: {
+            enabled: false
+          },
+          remote: {
+            enabled: true,
+            path: item.data.remote.path
+          }
+        },
+        assets: {
+          directory: {
+            enabled: false
+          },
+          remote: {
+            enabled: true,
+            path: item.data.remote.path
+          }
+        }
+      }
+
+      // Add the domain configuration.
+      DomainManager.addDomain(item.domain, configContent)
+    }
+
+    if (index === domains.length - 1) {
```

Comment:
Maybe I'm missing something, but it looks like we're dealing with synchronous code here, so instead of checking for `index` inside the `forEach`, you could simply run the `sendBackJSON` function after the `forEach`, when it's done iterating the array?

Reply:
true. it used to be somewhat more async